### PR TITLE
02 delete api endpoints

### PR DESCRIPTION
**Pull request by ldiu**
DELETE tag, DELETE attachment, and DELETE fact endpoints as well approval PUT. PUT is currently one way as it is not confirmed if an approved fact should be able to be unapproved.

### CODE REVIEW

File: app/handlers/factoid.js [hunk 1]
```diff
@@ -114,6 +113,17 @@ function filterFacts(facts, tags = []) {
     });
 }
 
+function deleteFactByID(factoidID) {
```
Comment (LukarioMC) [inline app/handlers/factoid.js]: Please add function documentation :)

File: app/handlers/factoid.js [hunk 2]
```diff
@@ -158,6 +168,18 @@ function addFact(factData) {
     }
 }
 
+function approveFactByID(factoidID) {
```
Comment (LukarioMC) [inline app/handlers/factoid.js]: Please add function documentation :)

File: app/routes/api.js [hunk 3]
```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
...
+});
+
+// API endpoint to delete a tag for a given factoid ID and category ID
+router.delete('/tag/:factoidID/:categoryID', (req, res) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const factoidID = req.params.factoidID;
+        const categoryID = req.params.categoryID;
+
+        const result = deleteTagforFactoid(parseInt(factoidID), parseInt(categoryID));
```
Comment (LukarioMC) [inline app/routes/api.js]: I feel like we should do this instead
```suggestion
        const factoidID = parseInt(req.params.factoidID);
        const categoryID = parseInt(req.params.categoryID);

		if (isNaN(factoidID) || isNaN(categoryID)) {
			return res.status(404).send({ message: 'Invalid fact or category in request.' });
		}

        const result = deleteTagforFactoid(factoidID, categoryID);
```

File: app/routes/api.js [hunk 4]
```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
...
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to delete a fact and associated tags and attachments
+router.delete('/fact/:factoidID', (req, res) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const factoidID = req.params.factoidID;
+
```
Comment (LukarioMC) [inline app/routes/api.js]: Pretty much same as above :)
```suggestion
        const factoidID = parseInt(req.params.factoidID);
        
		if (isNaN(factoidID)) {
			return res.status(404).send({ message: 'Invalid factoid ID in request.' });
		}
```

File: app/routes/api.js [hunk 5]
```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
...
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to approve a fact with the given id.
+router.put('/approve/:factoidID', (req, res, next) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        let factoidID = req.params.factoidID
+
```
Comment (LukarioMC) [inline app/routes/api.js]: Pretty much same as above :)
```suggestion
        const factoidID = parseInt(req.params.factoidID);
        
		if (isNaN(factoidID)) {
			return res.status(404).send({ message: 'Invalid factoid ID in request.' });
		}
```

File: app/routes/api.js [hunk 6]
```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
     res.redirect('back');
 });
 
+// API endpoint to delete an attachment for a given attachment ID
+router.delete('/attachment/:attachmentID', (req, res, next) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const attachmentID = req.params.attachmentID;
+
```
Comment (LukarioMC) [inline app/routes/api.js]: Pretty much same as above :)
```suggestion
        const attachmentID = parseInt(req.params.attachmentID);

		if (isNaN(attachmentID)) {
			return res.status(404).send({ message: 'Invalid attachment ID in request.' });
		}
```

File: app/handlers/factoid.js [hunk 7]
```diff
@@ -158,6 +168,18 @@ function addFact(factData) {
...
+    try {
+        let id = parseInt(factoidID)
+        let approveFactStmt = db.prepare('UPDATE factoid SET is_approved = 1 WHERE id = ?')
+        let result = approveFactStmt.run(id)
```
Comment (LukarioMC) [inline app/handlers/factoid.js]: Let the handlers above do type-checking and parsing, so you can just simply do:
```suggestion
        let approveFactStmt = db.prepare('UPDATE factoid SET is_approved = 1 WHERE id = ?')
        let result = approveFactStmt.run(factoidID)
```

**Review by LukarioMC — CHANGES_REQUESTED** [review]
Looking good so far, some changes please 👍

File: app/handlers/factoid.js (hunk 1 above)
Comment (ldiu) [reply app/handlers/factoid.js]: :+1:

File: app/handlers/factoid.js (hunk 2 above)
Comment (ldiu) [reply app/handlers/factoid.js]: :persevere:

**Review by LukarioMC — APPROVED** [review]
Looking good so far, approved and merging. 🥳

**Pull request merged**